resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/Client.scala

```diff
@@ -134,7 +140,16 @@ private[spark] class Client(
           .endSpec()
 
       val nonDriverPodKubernetesResources = mutable.Buffer[HasMetadata]()
-
+      val driverPodKubernetesCredentialsMounter = driverPodKubernetesCredentialsMounterProvider
+        .getDriverPodKubernetesCredentialsMounter(kubernetesAppId)
+      val driverPodKubernetesCredentialsSecret = driverPodKubernetesCredentialsMounter
+        .createCredentialsSecret()
+      val driverPodWithMountedDriverKubernetesCredentials = driverPodKubernetesCredentialsMounter
+        .mountDriverKubernetesCredentials(
+          basePod, driverContainer.getName, driverPodKubernetesCredentialsSecret)
+      nonDriverPodKubernetesResources ++= driverPodKubernetesCredentialsSecret.toSeq
+      val sparkConfWithDriverPodKubernetesCredentialLocations =
+        driverPodKubernetesCredentialsMounter.setDriverPodKubernetesCredentialLocations(sparkConf)
```

Comment:
I wonder if it's better to have this be all one method that returns a tuple of values, some new and some updated, and then apply those appropriately.  E.g. `(driverPodWithMountedCreds, sparkConfWithMountedCredLocations)`

Would that save any complexity here?